htdocs/core/lib/date.lib.php

```diff
@@ -1315,12 +1315,30 @@ function num_between_day($timestampStart, $timestampEnd, $lastday = 0)
  *	@param		int			$lastday            We include last day, 0: no, 1:yes
  *  @param		int			$halfday			Tag to define half day when holiday start and end
  *  @param      string		$country_code       Country code (company country code if not defined)
+ *  @param      int			$fk_user       		ID of the user for whom we are calculating
  *	@return    	int|string						Number of days or hours or string if error
  *  @seealso num_between_day(), num_public_holiday()
  */
-function num_open_day($timestampStart, $timestampEnd, $inhour = 0, $lastday = 0, $halfday = 0, $country_code = '')
+function num_open_day($timestampStart, $timestampEnd, $inhour = 0, $lastday = 0, $halfday = 0, $country_code = '', $fk_user = null)
```

Comment:
On last develop, we can already propagate directly the counrty id of user so we should be able to avoid to propagate the user->id.